### PR TITLE
Added dockerfile

### LINKED ISSUE #184: Add Docker container

### PR DESCRIPTION
**Pull request by Artur64**
resolves #184

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM elixir:1.9.4
```
Comment (thepiwo) [inline Dockerfile]: I think a more slim base-image can be used, e.g. the `alpine` variant. Maybe we should consider not sticking hard with `1.9.4` but `1.9`.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ENV LIBSODIUM_VER=1.0.16
+
+RUN apt update && \
+    apt install -y wget
+
+RUN mkdir -p libsodium-src && \
```
Comment (thepiwo) [inline Dockerfile]: the `alpine` linux package repository includes libsodium 1.0.18 as `libsodium-dev`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM elixir:1.9.4
+ENV LIBSODIUM_VER=1.0.16
+
+RUN apt update && \
```
Comment (thepiwo) [inline Dockerfile]: when sticking with a `debian` based image, this should use `apt-get` probably as the `apt` cli should not be used in scripts, as it suggests.

when switching to `alpine` a few more build dependencies are needed that are not installed by default
`RUN apk add --no-cache libsodium-dev git g++ make`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    apt install -y wget
+
+RUN mkdir -p libsodium-src && \
+    wget -O libsodium-src.tar.gz https://github.com/jedisct1/libsodium/releases/download/$LIBSODIUM_VER/libsodium-$LIBSODIUM_VER.tar.gz && tar -zxf libsodium-src.tar.gz -C libsodium-src --strip-components=1 && \
+    libsodium-src/configure && make -j$(nproc) && make install && ldconfig
+
+# lib folder should include generated low-level API modules
```
Comment (thepiwo) [inline Dockerfile]: I think the generation of low-level api modules should happen inside the docker build, as a previous step. (https://docs.docker.com/develop/develop-images/multistage-build/)

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+COPY config ./config
+COPY lib ./lib
+COPY mix.exs .
+COPY mix.lock .
+
+RUN mix local.hex --force && \
+    mix local.rebar --force && \
+    mix deps.get && mix compile
+
```
Comment (thepiwo) [inline Dockerfile]: cleanup of dependencies can happen here, e.g. for alpine `RUN apk del g++ make`

File: Dockerfile (hunk 4 above)
Comment (thepiwo) [reply Dockerfile]: maybe this can be a separate task, as it is not "blocking" this one

**Comment by thepiwo** [comment]
I added a PR https://github.com/aeternity/aepp-sdk-elixir/pull/186 to be merged into this one, with changes to use alpine as base

**Comment by thepiwo** [comment]
also the docker build could be part of the travis tests, or a separate stage in travis :)

File: README.md [hunk 6]
```diff
@@ -71,4 +71,13 @@ Module, which works with enoise protocol, which is used by Aeternity. Implemente
...
 
-**Full documentation can be found [here](http://aeternity.com/aepp-sdk-elixir/api-reference.html).**
\ No newline at end of file
+**Full documentation can be found [here](http://aeternity.com/aepp-sdk-elixir/api-reference.html).**
+
+## Use Docker image
+In order to play around with the **Elixir SDK**, you can use docker image:
+
+```
+docker pull danielapivanova/aepp-sdk-elixir:0.5.2
```
Comment (thepiwo) [inline README.md]: ask dincho to allow and push to the main aeternity repo

Comment (thepiwo) [reply README.md]: maybe add how to build as well :)

**Comment by DanielaIvanova** [comment]
@thepiwo Could you review please?

**Pull request merged**